packages/sanity/src/core/form/studio/inputResolver/fieldResolver.tsx

```diff
@@ -98,16 +138,37 @@ export function defaultResolveFieldComponent(
 
   const typeChain = getTypeChain(schemaType, new Set())
 
+  if (typeChain.some((t) => t.name === 'document')) {
+    return DocumentField as React.ComponentType<Omit<FieldProps, 'renderDefault'>>
+  }
+
   if (typeChain.some((t) => t.name === 'image' || t.name === 'file')) {
     return ImageOrFileField as React.ComponentType<Omit<FieldProps, 'renderDefault'>>
   }
+
   if (typeChain.some((t) => isReferenceSchemaType(t))) {
     return ReferenceField as React.ComponentType<Omit<FieldProps, 'renderDefault'>>
   }
 
+  if (isDescendentOfType(schemaType, 'slug')) {
+    return PrimitiveField as React.ComponentType<Omit<FieldProps, 'renderDefault'>>
+  }
```

Comment:
How does this change relate to this PR? If it is still required, could we use the same approach as above (eg using `typeChain`)?